kubeadm: use staticpod known tags for upgrade/init phases

pacoxu opened the pull request:
/kind bug
/release-note-none

See more in https://github.com/kubernetes/kubernetes/pull/110442#issuecomment-1150491847
```
time="21:01:54" level=debug msg="Running: docker exec kinder-patches-control-plane-1 /kinder/verify-patches.sh"
ERROR: /etc/kubernetes/manifests/etcd.yaml is not patched
ERROR: /etc/kubernetes/manifests/kube-apiserver.yaml is not patched
ERROR: /etc/kubernetes/manifests/kube-controller-manager.yaml is not patched
ERROR: /etc/kubernetes/manifests/kube-scheduler.yaml is not patched
```

fix https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-patches-latest

pacoxu committed:
kubeadm: use staticpod known tags for upgrade/init phases

pacoxu commented:
/release-note-none
/cc @neolit123

neolit123 reviewed (commented):
Can we instead just make all known targets to be returned by a function KnownTargets() in app/util/patches/patches.go ?

pacoxu commented:
> Can we instead just make all known targets to be returned by a function KnownTargets() in app/util/patches/patches.go ?

Updated. 
- I removed the cache replacement for known targets as well.

neolit123 reviewed (commented):
Thanks, i think this will fix it.
/lgtm
/approve

neolit123 commented:
/triage accepted
/priority important-soon

pacoxu commented:
/test pull-kubernetes-e2e-kind
